pkg/asset/machines/baremetal/machines.go

```diff
@@ -62,23 +66,32 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	return machines, nil
 }
 
-func provider(clusterName string, platform *baremetal.Platform, osImage string, userDataSecret string) *baremetalprovider.BareMetalMachineProviderSpec {
+func provider(clusterName string, platform *baremetal.Platform, osImage string, userDataSecret string) (*baremetalprovider.BareMetalMachineProviderSpec, error) {
 	// The rhcos-downloader container launched by the baremetal-operator downloads the image,
 	// compresses it to speed up deployments and makes it available on platform.ClusterProvisioningIP, via http
 	// osImage looks like:
-	//   https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/rhcos-4.2/42.80.20190725.1/rhcos-42.80.20190725.1-openstack.qcow2
+	//   https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/rhcos-4.2/42.80.20190725.1/rhcos-42.80.20190725.1-openstack.qcow2?sha256sum=123
 	// But the cached URL looks like:
 	//   http://172.22.0.3:6180/images/rhcos-42.80.20190725.1-openstack.qcow2/rhcos-42.80.20190725.1-compressed.qcow2
 	// See https://github.com/openshift/ironic-rhcos-downloader for more details
-	imageFilename := path.Base(osImage)
+	// The image is now formatted with a query string containing the sha256sum, we strip that here
+	// and it will be consumed for validation in ironic-rhcos-downloader
+	imageURL, err := url.Parse(osImage)
+	if err != nil {
+		return nil, errors.Wrap(err, "Invalid osImage format")
```

Comment:
nit: errors start with non-capitalized letter. it allows better wrapping.

Reply:
Done